src/command_modules/azure-cli-servicebus/azure/cli/command_modules/servicebus/custom.py

```diff
@@ -0,0 +1,444 @@
+        instance.action.requires_preprocessing = action_requires_preprocessing
+
+    if filter_sql_expression:
+        instance.sql_filter.sql_expression = filter_sql_expression
+
+    if filter_requires_preprocessing:
+        instance.sql_filter.requires_preprocessing = filter_requires_preprocessing
+
+    if correlation_id:
+        instance.correlation_filter.correlation_id = correlation_id
+
+    if to:
+        instance.correlation_filter.to = to
+
+    if message_id:
+        instance.correlation_filter.message_id = message_id
+
+    if reply_to:
+        instance.correlation_filter.reply_to = reply_to
+
+    if label:
+        instance.correlation_filter.label = label
+
+    if session_id:
+        instance.correlation_filter.session_id = session_id
+
+    if reply_to_session_id:
+        instance.correlation_filter.reply_to_session_id = reply_to_session_id
+
+    if content_type:
+        instance.correlation_filter.content_type = content_type
+
+    if requires_preprocessing:
+        instance.correlation_filter.requires_preprocessing = requires_preprocessing
+
+    return instance
+
+
+def empty_on_404(ex):
+    from azure.mgmt.servicebus.models import ErrorResponseException
+    if isinstance(ex, ErrorResponseException) and ex.response.status_code == 404:
+        return None
+    raise ex
+
+
+iso8601pattern = re.compile("^P(?!$)(\\d+Y)?(\\d+M)?(\\d+W)?(\\d+D)?(T(?=\\d)(\\d+H)?(\\d+M)?(\\d+.)?(\\d+S)?)?$")
+timedeltapattern = re.compile("^\\d+:\\d+:\\d+$")
+
+
+def return_valid_duration(objinstance, duration_property, update_value):
+    from datetime import timedelta
+    from isodate import parse_duration
+    if update_value is not None:
+        if iso8601pattern.match(update_value):
+            for attr, value in vars(objinstance).items():
+                if attr == duration_property:
+                    if parse_duration(update_value) < timedelta(days=10675199, minutes=10085, seconds=477581):
+                        return update_value
+
+                    if parse_duration(update_value) > timedelta(days=10675199, minutes=10085, seconds=477581):
+                        return None
+
+        if timedeltapattern.match(update_value):
+            day, miniute, seconds = update_value.split(":")
+            for attr, value in vars(objinstance).items():
+                if attr == duration_property:
+                    if timedelta(days=int(day), minutes=int(miniute), seconds=int(seconds)) < timedelta(days=10675199, minutes=10085, seconds=477581):
+                        return timedelta(days=int(day), minutes=int(miniute), seconds=int(seconds))
+
+                    if timedelta(days=int(day), minutes=int(miniute), seconds=int(seconds)) > timedelta(days=10675198, minutes=10085, seconds=477581):
+                        return None
+
+
+def return_valid_duration_create(update_value):
+    from datetime import timedelta
+    from isodate import parse_duration
+    from duration import to_iso8601
```

Comment:
You use `duration` for a single function. Is there another way you can achieve what you want without having this dependency?
It has a BSD license and hasn't been updated since 2016.
https://pypi.python.org/pypi/duration

Reply:
You didn't use this dependency in your extension so I wonder why you need it now.
https://github.com/Azure/azure-cli-extensions/blob/master/src/servicebus/setup.py#L27

Reply:
removed the duration as it was used for the testing and debug.